packages/ui/src/utils/index.tsx

```diff
@@ -83,3 +195,187 @@ export const getColorInTheme = (theme: any, color: string) => {
 
   return color;
 };
+
+export const __DEV__ = process.env.NODE_ENV !== 'production';
+
+export function runIfFn<T, U>(valueOrFn: T | ((...args: U[]) => T), ...args: U[]): T {
+  return isFunction(valueOrFn) ? valueOrFn(...args) : valueOrFn;
+}
+
+export function warn(options: { condition: boolean; message: string }) {
+  if (options.condition && __DEV__) {
+    console.warn(options.message);
+  }
+}
+
+export function error(options: { condition: boolean; message: string }) {
+  if (options.condition && __DEV__) {
+    console.error(options.message);
+  }
+}
+
+export type FunctionArguments<T extends Function> = T extends (...args: infer R) => any ? R : never;
+
+export function callAllHandlers<T extends (event: any) => void>(...fns: (T | undefined)[]) {
+  return function(event: FunctionArguments<T>[0]) {
+    fns.some(fn => {
+      fn && fn(event);
+      return event && event.defaultPrevented;
+    });
+  };
+}
+
+export type Merge<T1, T2> = Omit<T1, Extract<keyof T1, keyof T2>> & T2;
+
+export type SafeMerge<T, P> = P & Omit<T, keyof P>;
+
+export type UnionStringArray<T extends Readonly<string[]>> = T[number];
+
+export type Omit<T, K> = Pick<T, Exclude<keyof T, K>>;
```

Comment:
We don't need this, it comes with the language 
https://www.typescriptlang.org/docs/handbook/utility-types.html#omittk